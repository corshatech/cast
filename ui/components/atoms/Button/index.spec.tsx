/* Copyright 2023 Corsha.
   Licensed under the Apache License, Version 2.0 (the 'License');
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
   http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an 'AS IS' BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License. */

import renderer from 'react-test-renderer';
import { Button } from '.';

const cases: ['primary' | 'secondary', 'sm' | 'md' | 'lg'][] = [
  ['primary', 'sm'],
  ['primary', 'md'],
  ['primary', 'lg'],
  ['secondary', 'sm'],
  ['secondary', 'md'],
  ['secondary', 'lg'],
];

describe('Button', () => {
  test.each(cases)(
    'given %p as variant and %p as size, matches snapshot',
    (variant, size) => {
      const tree = renderer
        .create(
          <Button variant={variant} size={size}>
            Click Me
          </Button>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
