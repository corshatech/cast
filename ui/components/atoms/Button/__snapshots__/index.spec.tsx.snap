// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button given "primary" as variant and "lg" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-corsha-brand-blue text-white border-2 border-corsha-brand-blue hover:bg-corsha-brand-mid-blue hover:border-[#104457] focus:bg-corsha-brand-blue focus:border-2 focus:border-corsha-brand-green disabled:bg-[#A1A9B4] disabled:border-[#A1A9B4] disabled:cursor-not-allowed px-[75px] py-3.5"
>
  Click Me
</button>
`;

exports[`Button given "primary" as variant and "md" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-corsha-brand-blue text-white border-2 border-corsha-brand-blue hover:bg-corsha-brand-mid-blue hover:border-[#104457] focus:bg-corsha-brand-blue focus:border-2 focus:border-corsha-brand-green disabled:bg-[#A1A9B4] disabled:border-[#A1A9B4] disabled:cursor-not-allowed px-[70px] py-3"
>
  Click Me
</button>
`;

exports[`Button given "primary" as variant and "sm" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-corsha-brand-blue text-white border-2 border-corsha-brand-blue hover:bg-corsha-brand-mid-blue hover:border-[#104457] focus:bg-corsha-brand-blue focus:border-2 focus:border-corsha-brand-green disabled:bg-[#A1A9B4] disabled:border-[#A1A9B4] disabled:cursor-not-allowed px-10 py-2"
>
  Click Me
</button>
`;

exports[`Button given "secondary" as variant and "lg" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-white text-primary border-2 border-corsha-brand-blue hover:bg-highlight-blue focus:border-2 focus:border-corsha-brand-green disabled:border-[#A1A9B4] disabled:text-[#A1A9B4] disabled:hover:bg-white disabled:cursor-not-allowed px-[75px] py-3.5"
>
  Click Me
</button>
`;

exports[`Button given "secondary" as variant and "md" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-white text-primary border-2 border-corsha-brand-blue hover:bg-highlight-blue focus:border-2 focus:border-corsha-brand-green disabled:border-[#A1A9B4] disabled:text-[#A1A9B4] disabled:hover:bg-white disabled:cursor-not-allowed px-[70px] py-3"
>
  Click Me
</button>
`;

exports[`Button given "secondary" as variant and "sm" as size, matches snapshot 1`] = `
<button
  className="rounded-md space-x-3 leading-none inline-block bg-white text-primary border-2 border-corsha-brand-blue hover:bg-highlight-blue focus:border-2 focus:border-corsha-brand-green disabled:border-[#A1A9B4] disabled:text-[#A1A9B4] disabled:hover:bg-white disabled:cursor-not-allowed px-10 py-2"
>
  Click Me
</button>
`;
